clients/src/test/java/org/apache/kafka/common/security/JaasContextTest.java

```diff
@@ -68,11 +70,13 @@ public void tearDown() throws Exception {
 
     @Test
     public void testConfigNoOptions() throws Exception {
+        System.setProperty(ALLOWED_LOGIN_MODULES_CONFIG, "test.testConfigNoOptions");
```

Comment:
Adding this change means we break some compatibility ... 

1. print warning message if users define the `org.apache.kafka.disallowed.login.modules`
2. adopt the `org.apache.kafka.allowed.login.modules` if it is existent
3. evaluate the modules according to `DISALLOWED_LOGIN_MODULES_DEFAULT`

Additionally, we should open a jira for 5.0 to add `ALLOWED_LOGIN_MODULES_DEFAULT`

WDYT?